generators/generator-base-private.js

```diff
@@ -685,22 +697,27 @@ module.exports = class extends Generator {
      * Check if the generator specified as blueprint is installed.
      * @param {string} blueprint - generator name
      */
-    checkBlueprint(blueprint) {
+    checkBlueprint(blueprint, subGen = '') {
         if (blueprint === 'generator-jhipster') {
             this.error(`You cannot use ${chalk.yellow(blueprint)} as the blueprint.`);
         }
         const done = this.async();
         const localModule = path.join(process.cwd(), 'node_modules', blueprint);
-        if (!fs.existsSync(localModule)) {
-            shelljs.exec('yo --generators', { silent: true }, (err, stdout, stderr) => {
-                if (!stdout.includes(` ${blueprint}\n`) && !stdout.includes(` ${blueprint.replace('generator-', '')}\n`)) {
-                    this.error(`The ${chalk.yellow(blueprint)} blueprint provided is not installed. Please install it using command ${chalk.yellow(`npm i -g ${blueprint}`)}.`);
-                }
-                done();
-            });
-        } else {
+        if (fs.existsSync(localModule)) {
+            done();
+            return;
+        }
+        const generatorName = blueprint.replace('generator-', '');
+        if (this.env.get(`${generatorName}:${subGen}`)) {
```

Comment:
what is this for?

Reply:
This checks that the subgenerator is already registered and available in yeoman environment, since this is what yeoman will do internally in the composeWith here: https://github.com/jhipster/generator-jhipster/blob/master/generators/generator-base.js#L1753

This will mainly be the case with unit tests that uses yeoman-test .withGenerators().
checkBlueprint() is currently more restrictive than what composeExternalModule() can achieve.